test/api/test_plan_serialization_bwc.cpp

```diff
@@ -0,0 +1,71 @@
+#include "catch.hpp"
+#include "duckdb/common/field_writer.hpp"
+#include "duckdb/common/serializer/buffered_file_reader.hpp"
+#include "duckdb/common/serializer/buffered_file_writer.hpp"
+#include "duckdb/planner/planner.hpp"
+#include "duckdb/parser/parser.hpp"
+#include "test_helpers.hpp"
+#include "tpch-extension.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+string get_file_name(FileSystem &fs) {
+	auto file_name = string("test_plan_serialization_bwc.cpp");
+	auto path = string(__FILE__);
+	return path.replace(path.rfind(file_name), file_name.length(), "serialized_tpch_queries.binary");
+}
+
+TEST_CASE("Generate serialized TPCH file", "[.]") {
+	DuckDB db;
+	Connection con(db);
+	con.Query("CALL dbgen(sf=0.01)");
+	BufferedFileWriter serializer(db.GetFileSystem(), get_file_name(db.GetFileSystem()));
+	serializer.SetVersion(PLAN_SERIALIZATION_VERSION);
+	serializer.Write(serializer.GetVersion());
+	for (int q = 1; q < 22; q++) {
+		auto sql = TPCHExtension::GetQuery(q);
+
+		con.BeginTransaction();
+		Parser p;
+		p.ParseQuery(sql);
+
+		Planner planner(*con.context);
```

Comment:
We might want to consider turning these two tests into pragma functions in the future, so we can run e.g.:

```sql
PRAGMA export_logical_plan('count.binary', 'SELECT COUNT(*) FROM lineitem');
PRAGMA import_logical_plan('count.binary');
```

Reply:
Maybe I'm missing something but we'll also need some ways to close the file after writing multiple queries? Or are you thinking of having one file per query? Can you maybe sketch a bit how this will be used.

Reply:
I would say one serialization per file is fine

Reply:
happy to change. I assume it's ok to commit 30-ish binaries. One thing  I'm not clear on is how the import pragma will translate into a query that can be run. Can you unpack it a bit?

Reply:
The [from_substrait](https://github.com/duckdblabs/substrait/blob/main/src/substrait-extension.cpp#L148) function does something similar, perhaps have a look there?